2.5/alpine/entrypoint.sh

```diff
@@ -0,0 +1,13 @@
+#!/bin/sh
+set -e
+
+# Check if Caddyfile config is passed via $CADDYFILE
+if [[ "$CADDYFILE" ]]; then
+    echo 'Storing $CADDYFILE variable to ./Caddyfile'
+    printf "$CADDYFILE" > Caddyfile # echo doesn't preserve newlines
+fi
```

Comment:
Doesn't this run from the `WORKDIR` that is set to `/srv`? While the `--config` for `CMD` is set to `/etc/caddy/Caddyfile`? Is that intentional?

**NOTE:** the ENV check is using `[[ ... ]]`, this is [`bash` specific](https://stackoverflow.com/questions/669452/are-double-square-brackets-preferable-over-single-square-brackets-in-b/47576482#47576482) and not supported by plain `sh` AFAIK, if it's working it's probably due to `sh` being symlinked to an alternative shell (_likely `ash` for Alpine, but IIRC `sh` is limited, so this likely doesn't work?_).

Have you tested that this is working correctly? Surely you're changing the `CMD` on your end if that is the case?